nilearn/plotting/find_cuts.py

```diff
@@ -76,6 +76,9 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
         activation_threshold = fast_abs_percentile(my_map[my_map != 0].ravel(),
                                                    80)
     mask = np.abs(my_map) > activation_threshold - 1.e-15
+    # mask may be zero everywhere in rare cases
+    if (mask.max() == 0) and (mask.min() == 0):
```

Comment:
I actually double-check the claim that checking .max() and .min() was faster than using .any(), it seems to be true for float arrays but not boolean ones.


I'd be in favour of `if not mask.any()`

```python
import numpy as np
arr = np.random.random((64, 64, 64))
%timeit arr.max()
%timeit arr.any()
bool_arr = arr > 0.5
%timeit bool_arr.max()
%timeit bool_arr.any()
%timeit np.any(bool_arr)
```

```python
In [75]: import numpy as np

In [76]: arr = np.random.random((64, 64, 64))

In [77]: %timeit arr.max()
10000 loops, best of 3: 126 µs per loop

In [78]: %timeit arr.any()
1000 loops, best of 3: 868 µs per loop

In [79]: bool_arr = arr > 0.5

In [80]: %timeit bool_arr.max()
The slowest run took 6.30 times longer than the fastest. This could mean that an intermediate result is being cached 
100000 loops, best of 3: 9.85 µs per loop

In [81]: %timeit bool_arr.any()
The slowest run took 6.49 times longer than the fastest. This could mean that an intermediate result is being cached 
100000 loops, best of 3: 9.63 µs per loop

In [82]: %timeit np.any(bool_arr)
The slowest run took 5.46 times longer than the fastest. This could mean that an intermediate result is being cached 
100000 loops, best of 3: 14.1 µs per loop
```

Interestingly enough `np.any` has a 50% overhead compared to `.any`. While we are playing with micro-optimisations, `any(arr.flat)` could be a candidate but I guess it depends how many False you have to iterate through before finding the first True or the end of the array.